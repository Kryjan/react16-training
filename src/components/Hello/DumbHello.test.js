import React from "react";
import ReactDOM from "react-dom";
import { DumbHello } from "./DumbHello";

import { shallow, mount, render } from "enzyme";

describe("renders without crashing", () => {
    it("when default approach used", () => {
        const who = "Kryjan";
        const wrapper = render(<DumbHello who={who} onWhoChange={() => null} />);
        expect(wrapper.text).not.toBeNull();
        expect(wrapper.text()).toContain("Kryjan");

    });

    describe("when enzyme", () => {
        it("shallow used", () => {
            const who = "Kryjan";
            const wrapper = shallow(<DumbHello who={who} onWhoChange={() => null} />);
            expect(wrapper.text).not.toBeNull();
            expect(wrapper.text()).toContain("Kryjan");
        });

        it("mount used", () => {
            const who = "Kryjan";
            const wrapper = mount(<DumbHello who={who} onWhoChange={() => null} />);
            expect(wrapper.text).not.toBeNull();
            expect(wrapper.text()).toContain("Kryjan");
        });
    });

    describe("when enzyme snapshots taken with", () => {
        it("shallow", () => {
            const who = "Kryjan";

            expect(shallow(<DumbHello who={who} onWhoChange={() => null} />)).toMatchSnapshot();
        });

        it("mount", () => {
            const who = "Kryjan";

            expect(mount(<DumbHello who={who} onWhoChange={() => null} />)).toMatchSnapshot();
        });

        it("render", () => {
            const who = "Kryjan";

            expect(render(<DumbHello who={who} onWhoChange={() => null} />)).toMatchSnapshot();
        });
    });
});