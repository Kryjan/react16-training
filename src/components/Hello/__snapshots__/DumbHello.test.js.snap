// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing when enzyme snapshots taken with mount 1`] = `
<Component
  onWhoChange={[Function]}
  who="Kryjan"
>
  <div
    className="hello-container"
  >
    <span
      className="hello"
    >
      Hello 
      Kryjan
    </span>
    <input
      onChange={[Function]}
      type="text"
      value="Kryjan"
    />
  </div>
</Component>
`;

exports[`renders without crashing when enzyme snapshots taken with render 1`] = `
<div
  class="hello-container"
>
  <span
    class="hello"
  >
    Hello Kryjan
  </span>
  <input
    type="text"
    value="Kryjan"
  >
    
  </input>
</div>
`;

exports[`renders without crashing when enzyme snapshots taken with shallow 1`] = `
<div
  className="hello-container"
>
  <span
    className="hello"
  >
    Hello 
    Kryjan
  </span>
  <input
    onChange={[Function]}
    type="text"
    value="Kryjan"
  />
</div>
`;
